src/defrag.c

```diff
@@ -851,7 +851,7 @@ int defragLaterStep(redisDb *db, long long endtime) {
         }
 
         /* each time we enter this function we need to fetch the key from the dict again (if it still exists) */
-        dictEntry *de = dictFind(db->dict, defrag_later_current_key);
+        dictEntry *de = dictFind(getDict(db, defrag_later_current_key), defrag_later_current_key);
```

Comment:
i mean, if we do a "defrag later" step after each dict, rather than after each db, we'll maintain a shorter list of defrag later keys, and we won't need to do the calculateKeySlot overhead.

Reply:
Made the code change to handle defragLaterStep to be processed after each scan of slot.

Reply:
There is some issue with the current implementation causing a crash. Investigating.

Reply:
@oranagra please have a look at the implementation and let me know what you think about it.